experimental/envoy_proxy/README.md

```diff
@@ -7,8 +7,14 @@ in Envoy proxy.
 
 The experiment consists of 2 Envoy proxies: _client side_ and _server side_.
 
-- Client side proxy accepts TCP streams and tunnels them over HTTP streams.
-- Server side proxy listens for HTTP streams and unwraps TCP streams from them.
+- Client side proxy:
+  - Listens for TCP connections on a local port
+  - Accepts TCP streams and tunnels them over HTTP streams
+    - Each individual TCP connection is tunnelled over a dedicated HTTP stream
+      (using POST requests)
+- Server side proxy:
+  - Listens for HTTP streams and unwraps TCP streams from them
+  - Forwards TCP streams to the application server
```

Comment:
I think this part is still a bit verbose / ambiguous. Since the 1:many or many:many relationship is important, I suggest always wording these sentences as `for each X it does Y`, like a `for` loop, so it's clear the 1:many or 1:1 relationship. If you just write `it does Y on the Xs` it's not clear if it does it on each of them, or on all of them at once, so I tend to avoid this wording if possible.